Document the manual approach how to use the Rich error model (#1507)
akka/akka-grpc: docs/src/main/paradox/client/details.md

Java
:  @@snip [GreeterClient.java](/plugin-tester-java/src/main/java/example/myapp/helloworld/LiftedGreeterClient.java) { #with-metadata }

## Rich error model

Beyond status codes you can also use the [Rich error model](https://www.grpc.io/docs/guides/error/#richer-error-model). Currently there is no native support for this concept in Akka gRPC. However you can use the following manual approach.

> Please follow the @ref[server section](../server/details.md) how to add the necessary dependency.

Reviewer: Might be easier to follow just to repeat the instruction here?

Java
:  @@snip [GreeterClient.java](/plugin-tester-java/src/main/java/example/myapp/helloworld/LiftedGreeterClient.java) { #with-metadata }

## Rich error model

> Beyond status codes you can also use the @link:[Rich error model](https://www.grpc.io/docs/guides/error/#richer-error-model) { open=new }. Currently there is no native support for this concept in akka-grpc. However you can use the following manual approach.

Reviewer: I think we usually just use 'regular' markdown links:

```suggestion
Beyond status codes you can also use the [Rich error model](https://www.grpc.io/docs/guides/error/#richer-error-model). Currently there is no native support for this concept in akka-grpc. However you can use the following manual approach.
```

Reviewer: I think we prefer 'Akka gRPC' in running text, and `akka-grpc` in places like URL's and artifact names:

```suggestion
Beyond status codes you can also use the @link:[Rich error model](https://www.grpc.io/docs/guides/error/#richer-error-model) { open=new }. Currently there is no native support for this concept in Akka gRPC. However you can use the following manual approach.
```

Please follow @ref[this](../server/details.md) how to add the necessary dependency.

> Make the request and cast it as a `StatusRuntimeException`:

Reviewer: I think we shouldn't recommend casting, as we can't be sure it'll always be a `StatusRuntimeException` - instead we could show how to check whether it's a StatusRuntimeException, and only in that case look for rich details.

Author: Jea that makes sense.

## Rich error model

Beyond status codes you can also use the [Rich error model](https://www.grpc.io/docs/guides/error/#richer-error-model). Currently there is no native support for this concept in Akka gRPC. However you can use the following manual approach.

Add the following dependency to receive required classes (that are based on the [common protobuf](https://cloud.google.com/apis/design/errors#error_model)):

`sbt`
:   @@@vars
```sbt
libraryDependencies += "io.grpc" % "grpc-protobuf" % "1.42.1"
```
@@@

`gradle`
:   @@@vars
```gradle
dependencies {
     implementation 'io.grpc:grpc-protobuf:1.42.1'
}
```
@@@

`maven`
:   @@@vars
```maven
<dependency>
      <groupId>io.grpc</groupId>
      <artifactId>grpc-protobuf</artifactId>
      <version>1.42.1</version>
</dependency>
```
@@@

Extract the `StatusRuntimeException` and parse the Rich error model to access `code`, `message` and `details`.

Scala
:  @@snip [GreeterClient.scala](/interop-tests/src/test/scala/akka/grpc/scaladsl/RichErrorModelSpec.scala) { #client_request }

Java
:  @@snip[RichErrorModelSpec](/interop-tests/src/test/java/example/myapp/helloworld/grpc/RichErrorModelTest.java) { #client_request }

> Please look @ref[here](../server/details.md) how to handle this on the server.

Reviewer: Good to cross-reference, maybe a bit more specific?

```suggestion
Please look @ref[here](../server/details.md) how to create errors with such details on the server side.
```
